drivers/can/can_mcan.c

```diff
@@ -1469,6 +1469,23 @@ int can_mcan_init(const struct device *dev)
 		return err;
 	}
 
+#ifdef CONFIG_CAN_RX_TIMESTAMP
+	/*
+	 * Enable the internal timestamp counter by default. SoC-specific driver frontends can
+	 * overwrite this if configured for using a SoC-specific, external timestamp counter.
+	 */
+	reg = FIELD_PREP(CAN_MCAN_TSCC_TCP, config->timestamp_prescaler - 1U) |
+		FIELD_PREP(CAN_MCAN_TSCC_TSS, 1U);
```

Comment:
You could consider adding defines for the TSCV source selected with TSCC.TSS, as multiple platform drivers will change it to the external counter, but not blocking.